presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HistoryTable.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.metadata.InternalTable;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorPageSource;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.FixedPageSource;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.predicate.TupleDomain;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+
+public class HistoryTable
+        implements SystemTable
+{
+    private final IcebergTableHandle tableHandle;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = ImmutableList.of(
+            new ColumnMetadata("made_current_at", TIMESTAMP),
```

Comment:
Good point, but I looked into this and it turns out to be legit to use `TIMESTAMP` here. The timestamp values are populated by Iceberg using [`System#currentTimeMillis()`](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#currentTimeMillis), which is "the difference between the current time and midnight, January 1, 1970 UTC".

1. When using `TIMESTAMP`, values are rendered in the session time zone. (`SqlTimestamp`)
2. When using `TIMESTAMP_WITH_TIME_ZONE`, `SqlTimestampWithTimeZone` will be used to parse the timestamp values. But these values are not encoded in the Presto way and will incur an error when being parsed.

Reply:
This is about semantics, not encoding. You are correct that `TIMESTAMP` is rendered in the session time zone today. However, that is wrong and we are fixing it: https://prestosql.io/docs/current/language/timestamp.html

The `TIMESTAMP` type is simply a wall clock time (date and time fields). `TIMESTAMP WITH TIME ZONE` represents a specific point in time in a particular time zone. Thus, the latter seems more appropriate here, since the column is the exact time when the table was updated. For time zone, using the session time zone here seems appropriate. Think of this like the `current_timestamp` function which returns the current time in the session time zone.

Of course, we would need to properly encode the value when returning the column.